### PR TITLE
allow quantity to be empty when using autoscaling

### PR DESCRIPTION
**Pull request by thecodeassassin**

### CODE REVIEW

**Review by Styren — COMMENTED** [review]
Will this throw an error on the client if quantity is missing and autoscaling is disabled? If not I think we should do that, so we don't get a confusing 400 error.

**Comment by thecodeassassin** [comment]
> Will this throw an error on the client if quantity is missing and autoscaling is disabled? If not I think we should do that, so we don't get a confusing 400 error.

It already does that.

**Comment by Styren** [comment]
> > Will this throw an error on the client if quantity is missing and autoscaling is disabled? If not I think we should do that, so we don't get a confusing 400 error.
> 
> It already does that.

Sorry, can you show me where? I can't find it in the code.

File: symbiosis/resource_node_pool.go [hunk 1]
```diff
@@ -113,6 +114,21 @@ func ResourceNodePool() *schema.Resource {
 		UpdateContext: resourceNodePoolUpdate,
 		DeleteContext: resourceNodePoolDelete,
 		Schema:        resourceSchema,
+		CustomizeDiff: func(ctx context.Context, d *schema.ResourceDiff, i interface{}) error {
```
Comment (thecodeassassin) [inline symbiosis/resource_node_pool.go:117]: @Styren here

**Comment by thecodeassassin** [comment]
> > > Will this throw an error on the client if quantity is missing and autoscaling is disabled? If not I think we should do that, so we don't get a confusing 400 error.
> 
> > 
> 
> > It already does that.
> 
> 
> 
> Sorry, can you show me where? I can't find it in the code.

https://github.com/symbiosis-cloud/terraform-provider-symbiosis/pull/20#discussion_r1113939347

**Review by Styren — APPROVED** [review]
Ah alright I thought this didn't evaluate to true if autoscaling isn't defined but I guess it does?
```
			if d.HasChange("autoscaling") {
```

**Review by Styren — APPROVED** [review]

**Pull request merged**